api/v1alpha3/azurecluster_conversion.go

```diff
@@ -195,10 +209,10 @@ func Convert_v1alpha4_SecurityGroup_To_v1alpha3_SecurityGroup(in *infrav1alpha4.
 	out.ID = in.ID
 	out.Name = in.Name
 
-	out.IngressRules = make(IngressRules, len(in.IngressRules))
-	for i := range in.IngressRules {
+	out.IngressRules = make(IngressRules, len(in.SecurityRules))
+	for i := range in.SecurityRules {
 		out.IngressRules[i] = IngressRule{}
```

Comment:
Shouldn't we check for `securityRule.Direction == Inbound` here?

Reply:
so this causes the fuzzy conversion test to fail, which I guess makes sense since we're dropping rules 🤔  the issue is that on round-trip conversion the entire rule won't be restored

Reply:
ok I think I fixed it & and improved the conversion. @shysank I kept the last commit separate to make it easier to review, let me know what you think about the conversion before I squash.

Reply:
Thanks @CecileRobertMichon for fixing the conversion and adding comments! I've just added one small feedback to the conversion code, ptal.